In [ ]:
# Toma una muestra aleatoria simple de tamaño 30

In [1]:
# Importar base de datos
import pandas as pd
df = pd.read_csv('social_media_engagement.csv')
df.head()

user_id  followers   avg_likes  time_in_app_min
0        1      15895  310.875395       145.611740
1        2        960  397.293564       127.089777
2        3      76920  334.581034       102.879860
3        4      54986  364.307423       142.026547
4        5       6365  340.240159       101.501848

In [2]:
# Tamaño de la poblacion
N = df.shape[0]
print(f'Tamaño de la población: {N}')

Tamaño de la población: 1500


In [3]:
#Muestra aleatoria simple de tamaño 120
n = 120
muestra = df.sample(n=n, replace=False, random_state=42)
muestra.head()

user_id  followers   avg_likes  time_in_app_min
1116     1117      12054  297.971409       145.225163
1368     1369      85614  373.516039       135.141371
422       423      93948  328.964507       113.861347
413       414      76529  266.894374       117.962558
451       452      72715  222.523437       139.571661

In [4]:
#Internalo de confianza al 95% para el promedio de minutos diarios en la app time_in_app_min
import numpy as np
from scipy.stats import t

# Media poblacional
media_pob_min = df['time_in_app_min'].mean()
# Media y desviación en la muestra
media_muestra_min = muestra['time_in_app_min'].mean()
s_muestra_min = muestra['time_in_app_min'].std(ddof=1)

# Calcular IC
SE_math = s_muestra_min / np.sqrt(n)
t_critico = t.ppf(0.975, df=n-1)
IC_inf_math = media_muestra_min - t_critico * SE_math
IC_sup_math = media_muestra_min + t_critico * SE_math

print(f'Media horas: {media_pob_min:.2f}')
print(f'IC 95% para los minutos en la app: ({IC_inf_math:.2f}, {IC_sup_math:.2f})')



Media horas: 119.79
IC 95% para los minutos en la app: (110.50, 121.35)


In [5]:
#Internalo de confianza al 95% para el promedio de likes avg_likes
import numpy as np
from scipy.stats import t

# Media poblacional
media_pob_likes = df['avg_likes'].mean()
# Media y
media_muestra_likes = muestra['avg_likes'].mean()
s_muestra_likes = muestra['avg_likes'].std(ddof=1)

#
SE_math = s_muestra_likes / np.sqrt(n)
t_critico = t.ppf(0.975, df=n-1)
IC_inf_math = media_muestra_likes - t_critico * SE_math
IC_sup_math = media_muestra_likes + t_critico * SE_math

print(f'Media likes: {media_pob_likes:.2f}')
print(f'IC 95% para los likes: ({IC_inf_math:.2f}, {IC_sup_math:.2f})')



Media likes: 302.59
IC 95% para los likes: (292.33, 310.03)


In [6]:
# Compara las medias de las muestras con las medias poblacionales. ¿Las estimaciones parecen precisas?
print(f'Media poblacional de minutos en la app: {media_pob_min:.2f}')
print(f'Media muestral de minutos en la app: {media_muestra_min:.2f}')
print(f'Media poblacional de likes: {media_pob_likes:.2f}')
print(f'Media muestral de likes: {media_muestra_likes:.2f}')


Media poblacional de minutos en la app: 119.79
Media muestral de minutos en la app: 115.93
Media poblacional de likes: 302.59
Media muestral de likes: 301.18


Los datos nos muestran las medias poblacionales y muestrales para dos métricas: minutos en la aplicación y "me gusta". Para los minutos en la aplicación, la media poblacional es de 119.79 minutos, mientras que la media muestral es de 115.93 minutos. En cuanto a los "me gusta", la media poblacional es de 302.59, y la media muestral es de 301.18. En ambos casos, se observa que la media muestral es ligeramente inferior a la media poblacional, lo cual es un comportamiento esperado cuando se utiliza una muestra para estimar las características de una población más grande.

In [7]:
#Realiza 300 muestras aleatorias de tamaño 40 para construir la distribucion muestral del promedio de likes. Calcula la media y desviacion de esa distribucion.

import numpy as np
#Realiza 300 muestras aleatorias de tamaño 40 para construir la distribucion muestral del promedio de likes. Calcula la media y desviacion de esa distribucion.
np.random.seed(42) # Para reproducibilidad
num_muestras = 300
tamano_muestra = 40
medias_muestrales_likes = []

for _ in range(num_muestras):
  muestra_likes = df['avg_likes'].sample(n=tamano_muestra, replace=False)
  media_muestra_likes_sim = muestra_likes.mean()
  medias_muestrales_likes.append(media_muestra_likes_sim)

# Calcula la media y desviación de la distribución muestral
media_distribucion_muestral = np.mean(medias_muestrales_likes)
desviacion_distribucion_muestral = np.std(medias_muestrales_likes, ddof=1)

print(f'Media de la distribución muestral del promedio de likes: {media_distribucion_muestral:.2f}')
print(f'Desviación de la distribución muestral del promedio de likes: {desviacion_distribucion_muestral:.2f}')

Media de la distribución muestral del promedio de likes: 302.59
Desviación de la distribución muestral del promedio de likes: 8.00


¿Cómo evidencia esta simulación el Teorema Central del Límite?
¿Qué le dirías a un gerente de marketing que quiere saber cuántos likes promedio esperar en un usuario típico?"

El código proporcionado no simula directamente el Teorema Central del Límite (TCL) de manera explícita mediante la visualización de la distribución de las medias muestrales. Sin embargo, el proceso de tomar múltiples muestras y calcular sus medias para luego analizar la distribución de estas medias (`medias_muestrales_likes`) **es el núcleo de lo que el TCL describe**.

Aquí te explico cómo esta parte del código **evidencia** el TCL, incluso sin una visualización explícita:

1.  **Generación de Múltiples Muestras:** El código toma 300 muestras aleatorias de tamaño 40 de la variable `avg_likes`.
2.  **Cálculo de las Medias Muestrales:** Para cada una de estas 300 muestras, calcula la media de `avg_likes`.
3.  **Distribución de las Medias:** Al almacenar estas 300 medias muestrales en la lista `medias_muestrales_likes`, estamos creando una aproximación de la **distribución muestral de la media**.

**¿Cómo evidencia esto el TCL?**

El TCL establece que si tomas muestras lo suficientemente grandes (generalmente se considera n >= 30) de cualquier población, independientemente de su distribución original, la distribución de las medias muestrales se aproximará a una **distribución normal**.

Aunque el código no grafica esta distribución, el hecho de que estés calculando la media y desviación estándar de `medias_muestrales_likes` (que son la media y desviación estándar de la distribución muestral) te permite hacer inferencias basadas en el TCL. Si graficaras esta lista, verías una forma que se parecería a una campana (distribución normal), y la media de esta distribución muestral sería muy cercana a la media poblacional de `avg_likes`. La desviación estándar de esta distribución muestral es una estimación del error estándar de la media.

**En resumen:** La simulación, al generar y recopilar las medias de múltiples muestras, está creando los datos que, según el TCL, deben seguir una distribución normal. Aunque la normalidad no se *muestra* directamente, el proceso de calcular la media y desviación de `medias_muestrales_likes` se basa en la expectativa que el TCL nos da sobre cómo se comportan estas medias.

---

**¿Qué le dirías a un gerente de marketing que quiere saber cuántos likes promedio esperar en un usuario típico?**

Le diría lo siguiente, basándome en el análisis realizado en el código:

"Basado en nuestro análisis de los datos disponibles, podemos estimar que el **número promedio de likes que un usuario típico recibe está en el rango de [valor del límite inferior del IC de likes] a [valor del límite superior del IC de likes] con un 95% de confianza**.

Esto significa que si tomáramos muchas muestras similares de usuarios, esperaríamos que el verdadero promedio de likes en la población caiga dentro de este intervalo en el 95% de las veces.

Para ponerlo de forma más sencilla: aunque el promedio exacto para todos nuestros usuarios es desconocido, nuestra mejor estimación puntual es de [media muestral de likes] likes, y estamos bastante seguros (un 95% de seguridad) de que el promedio real se encuentra en ese rango.

Este intervalo nos da una idea de la variabilidad y la precisión de nuestra estimación. Es una forma más robusta de informar un promedio que simplemente dar un único número, ya que reconoce que nuestra estimación proviene de una muestra y, por lo tanto, tiene cierta incertidumbre."

**Detalles a considerar para el gerente de marketing:**

*   **Lenguaje claro:** Evitar jerga técnica. "Rango" e "intervalo de confianza" son más fáciles de entender que "IC".
*   **Implicaciones:** Explicar qué significa el 95% de confianza en términos prácticos.
*   **Contexto:** Relacionarlo con la "usuaria típica" que preguntó el gerente.
*   **Limitaciones:** Podrías añadir que esta es una estimación basada en los datos actuales y el tamaño de la muestra utilizada.

Para dar la respuesta específica, necesitarías reemplazar `[valor del límite inferior del IC de likes]` y `[valor del límite superior del IC de likes]` con los valores calculados por tu código para `IC_inf_math` y `IC_sup_math` de los likes.